adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/impl/VertxBasedAmqpProtocolAdapter.java

```diff
@@ -762,18 +749,29 @@ protected void handleRemoteSenderOpenForCommands(
         });
     }
 
+    private void onCommandConnectionClose(final String tenantId, final String deviceId, final Device authenticatedDevice,
+            final ProtocolAdapterCommandConsumer consumer, final OptionalInt traceSamplingPriority) {
+
+        final Span span = newSpan("handle closing of command connection", authenticatedDevice, traceSamplingPriority);
+        final Future<ProtonDelivery> sendEventFuture = sendDisconnectedTtdEvent(tenantId, deviceId, authenticatedDevice,
+                span.context());
+        final Future<Void> closeConsumerFuture = consumer.close(span.context());
+
+        CompositeFuture.join(sendEventFuture, closeConsumerFuture).recover(thr -> {
+            Tags.ERROR.set(span, true);
```

Comment:
why not log the error to the span?

Reply:
The error(s) will be logged at the respective child spans. Therefore I think it's not needed here, also because both of the child operations could have failed, meaning there are 2 errors.